### PR TITLE
PAT-2139: Security Upgrade for H2 etc various repos

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 plugins {
-  id("uk.gov.justice.hmpps.gradle-spring-boot") version "3.3.16"
-  kotlin("plugin.spring") version "1.5.31"
-  kotlin("plugin.jpa") version "1.5.31"
+  id("uk.gov.justice.hmpps.gradle-spring-boot") version "4.0.3"
```
Comment (petergphillips) [inline build.gradle.kts:2]: could have gone to `4.0.4`

**Review by andymarke — APPROVED** [review]
LGTM

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-bin.zip
```
Comment (petergphillips) [inline gradle/wrapper/gradle-wrapper.properties:3]: `7.4` is out too`

File: src/main/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/NotificationController.kt [hunk 3]
```diff
@@ -16,15 +16,17 @@ import uk.gov.justice.digital.hmpps.cmd.api.service.NotificationService
 import java.time.LocalDate
 import java.util.Optional
 
-@Api(tags = ["notifications"])
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/NotificationController.kt:19]: that could have been converted to `@Tag` instead

File: build.gradle.kts (hunk 1 above)
Comment (steverendell) [reply build.gradle.kts:2]: wow ok, moving 🎯 !

File: src/main/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/ShiftController.kt [hunk 4]
```diff
@@ -16,15 +16,17 @@ import uk.gov.justice.digital.hmpps.cmd.api.service.ShiftService
...
-  @ApiOperation(value = "Retrieve all details for a user between two dates")
-  @ApiResponses(
-    value = [
-      ApiResponse(code = 200, message = "OK")
+  @Operation(
+    summary = "Retrieve all details for a user between two dates",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ShiftDto::class))]
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/ShiftController.kt:28]: again this looks wrong, should be `ArraySchema` or better yet, just remove

File: src/main/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/ShiftController.kt [hunk 5]
```diff
@@ -16,15 +16,17 @@ import uk.gov.justice.digital.hmpps.cmd.api.service.ShiftService
 import java.time.LocalDate
 import java.util.Optional
 
-@Api(tags = ["shift controller"])
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/cmd/api/controllers/ShiftController.kt:19]: could have gone with `@Tag`

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (steverendell) [reply gradle/wrapper/gradle-wrapper.properties:3]: 👍

File: .circleci/config.yml [hunk 6]
```diff
@@ -1,17 +1,15 @@
...
+  hmpps: ministryofjustice/hmpps@3.12
 
 jobs:
   validate:
-    executor: validator
+    executor:
+      name: hmpps/java
+      tag: "17.0"
+    environment:
+      _JAVA_OPTIONS: -Xmx500m -XX:ParallelGCThreads=2 -XX:ConcGCThreads=2 -Djava.util.concurrent.ForkJoinPool.common.parallelism=2 -Dorg.gradle.daemon=false -Dkotlin.compiler.execution.strategy=in-process
```
Comment (steverendell) [inline .circleci/config.yml:12]: 256m wasnt enough 🤷

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**